### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jfaltermeier**
* rollup
* vite
* body-parser
* express
* path-to-regexp
* send
* serve-static
* micromatch
* webpack
* cookie
* junit in demos
* update workflow actions (use sha for non github actions)
* update helm dependencies to latest version (but keycloak)

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: node/landing-page/package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "landing-page",
   "version": "0.1.0",
+  "license": "EPL-2.0",
```
Comment (lucas-koehler) [inline node/landing-page/package.json:4]: I think our license actually is the following. We also use this in headers in the operator java files and the LICENSE file. WDYT?
```suggestion
  "license": "EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0",
```

File: node/testing-page/package.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "testing-page",
   "version": "0.1.0",
+  "license": "EPL-2.0",
```
Comment (lucas-koehler) [inline node/testing-page/package.json:4]: I think our license actually is the following. We also use this in headers in the operator java files and the LICENSE file. WDYT?
```suggestion
  "license": "EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0",
```

**Review by lucas-koehler — COMMENTED** [review]
The changes already look quite good to me :)
I have 2 small remarks inline regarding the license.

I wonder whether we should also update the maven dependencies in https://github.com/eclipse-theia/theia-cloud/blob/main/java/common/maven-conf/pom.xml
I.e. quarkus, maven surefire, etc. Was this missed or left out on purpose?

Note that I did not test this as the changes are pretty small. Let me know if I should.

**Comment by jfaltermeier** [comment]
> The changes already look quite good to me :) I have 2 small remarks inline regarding the license.
> 
> I wonder whether we should also update the maven dependencies in https://github.com/eclipse-theia/theia-cloud/blob/main/java/common/maven-conf/pom.xml I.e. quarkus, maven surefire, etc. Was this missed or left out on purpose?
> 
> Note that I did not test this as the changes are pretty small. Let me know if I should.

I checked for a new version of https://mvnrepository.com/artifact/io.javaoperatorsdk/kubernetes-webhooks-framework-core at the beginning of November. I usually pick the corresponding io.fabric8 k8s-client version from this and update Quarkus accordingly, ensuring all dependencies are built with the same or a similar k8s-client version.

Since there wasn't a release at this time, I mainly focused on Dependabot alerts for Maven.

There’s a new version available now, so I’ll update to it after 1.0.

**Review by lucas-koehler — APPROVED** [review]
Thanks for the update and explanation. LGTM :sparkles:

**Pull request merged**